crowbar_framework/app/assets/javascripts/installer.js

```diff
@@ -0,0 +1,45 @@
+$(document).ready(function() {
+  function statusCheck() {
+    $.ajax({
+      type: "GET",
+      dataType: "json",
+      url: "/installer/status.json",
+      success: function(data)
+      {
+        var failed = data.failed;
+        data = data.steps;
+        var mostRecent = $("li." + data[data.length-1]);
+        var mostRecentIcon = mostRecent.children();
+
+        $.each(data, function (index, value) {
+          $("li." + value).removeClass("list-group-item-success");
+          $("li." + value).children("span").removeClass("fa-hourglass-o fa-circle-o-notch fa-spin");
+          $("li." + value).children("span").addClass("fa-check");
+        });
+
+        mostRecentIcon.removeClass(function (index, css) {
+          match = css.match(/^fa-/g);
+          if (match !== null) {
+            return match.join("");
+          }
+        });
+        mostRecentIcon.addClass("fa-check");
+
+        if (failed) {
+          mostRecent.next("li").addClass("list-group-item-danger");
```

Comment:
Something like this should work and there is no need to lookup the elements again and again:

```
mostRecent
  .next("li")
  .addClass("list-group-item-danger")
  .find('span')
  .removeClass("fa-hourglass-o fa-circle-o-notch fa-spin")
  .addClass("fa-times");
```